nemo/collections/asr/models/ctc_models.py

```diff
@@ -123,18 +126,22 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
 
     def __init__(
         self,
-        preprocessor_params: Dict,
-        encoder_params: Dict,
-        decoder_params: Dict,
-        spec_augment_params: Optional[Dict] = None,
+        preprocessor_config: DictConfig,
+        encoder_config: DictConfig,
+        decoder_config: DictConfig,
+        spec_augment_config: Optional[DictConfig] = None,
     ):
         super().__init__()
-        self.preprocessor = Serialization.from_config_dict(preprocessor_params)
-        self.encoder = Serialization.from_config_dict(encoder_params)
-        self.decoder = Serialization.from_config_dict(decoder_params)
+        # self.preprocessor = Serialization.from_config_dict(preprocessor_config)
+        self.preprocessor = hydra.utils.instantiate(preprocessor_config)
+        # self.encoder = Serialization.from_config_dict(encoder_config)
+        self.encoder = hydra.utils.instantiate(encoder_config)
+        # self.decoder = Serialization.from_config_dict(decoder_config)
```

Comment:
@okuchaiev  I honestly do not understand why you do not want to create modules by names... IMO this is less readable and more error prone. 

Just a suggestion thou.